client/js/editmode.js

```diff
@@ -1,8 +1,16 @@
 let edit = false;
 
+function generateUniqueWidgetID() {
+  let id;
+  do {
+    id = Math.random().toString(36).substring(3, 7);
+  } while (widgets.has(id));
```

Comment:
If the test function is a parameter, this could also be used for other cases.

Reply:
Both the test function and the specifics of the random function could be parameters. But given that there was only one other place in the code that potentially needed fixing (in states) I decided not to go that route. (Also it would have required learning lots more JS that I don't know :) )

Reply:
No problem, I'm just saying once we add the check for other stuff, it could work like that.

There are also a few of those in the server side code.